internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go

```diff
@@ -761,9 +764,55 @@ func (s *service) ListContainer(ctx context.Context, req *provider.ListContainer
 
 	if isVirtualRoot(req.Ref.ResourceId) {
 		// The root is empty, it is filled by mountpoints
+		// but when accessing the root via /dav/spaces we need to list the content
+
+		receivedShares, _, err := s.fetchShares(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "sharesstorageprovider: error calling ListReceivedSharesRequest")
+		}
+
+		infos := []*provider.ResourceInfo{}
+		for _, share := range receivedShares {
+			if share.GetState() != collaboration.ShareState_SHARE_STATE_ACCEPTED {
+				continue
+			}
+
+			statRes, err := s.gateway.Stat(ctx, &provider.StatRequest{
+				Opaque: req.Opaque,
+				Ref: &provider.Reference{
+					ResourceId: share.Share.ResourceId,
+					// Path: "." TODO force a relative request? should not matter because we always only want the name
```

Comment:
Since it is containing an ID pointing to the correct resource each storageprovider should be able to handle it. The path is irrelevant if the ID already points to the correct resource.

Anyways the comment here is not very helpful. Either force a relative request now or we remove the commented line.